pennylane/templates/subroutines/qdrift.py

```diff
@@ -0,0 +1,257 @@
+    of a given Hamiltonian.
+
+    # TODO: link paper reference, mention that we assume each operator is normalized.
+
+    The QDrift subroutine provides a method to approximate the matrix exponential of a Hamiltonian
+    expressed as a linear combination of terms which in general do not commute. For the Hamiltonian
+    :math:`H = \Sigma_j h_j H_{j}`, the product formula is constructed by random sampling from the
+    terms of the Hamiltonian with the probability :math:`p_j = h_j / \sum_{j} hj` as:
+
+    .. math::
+
+        \prod_{j}^{n} e^{i \lambda H_j \tau / n},
+
+    where :math:`\tau` is time, :math:`\lambda = \sum_j |h_j|` and :math:`n` is the total number of
+    terms to be added to the product.
+
+    Args:
+        hamiltonian (Union[~.Hamiltonian, ~.Sum]): The Hamiltonian written as a sum of operations
+        time (int or float): The time of evolution, namely the parameter :math:`t` in :math:`e^{-iHt}`
+        n (int): An integer representing the number of exponentiated terms
+        seed (int): The seed for the random number generator
+
+    Raises:
+        TypeError: The ``hamiltonian`` is not of type :class:`~.Hamiltonian`, or :class:`~.Sum`
+        QuantumFunctionError: If the coefficients of ``hamiltonian`` are trainable and are used
+            in a differentiable workflow.
+
+    **Example**
+
+    .. code-block:: python3
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.PauliX(0), qml.PauliZ(0)]
+        H = qml.dot(coeffs, ops)
+
+        dev = qml.device("default.qubit", wires=2)
+        @qml.qnode(dev)
+        def my_circ():
+            # Prepare some state
+            qml.Hadamard(0)
+
+            # Evolve according to H
+            qml.QDrift(H, time=1.2, n = 10)
+
+            # Measure some quantity
+            return qml.probs()
+
+    >>> my_circ()
+    [0.71061676 0.         0.28938324 0.        ]
+
+
+    .. details::
+        :title: Usage Details
+
+        We currently **Do NOT** support computing gradients with respect to the
+        coefficients of the input Hamiltonian. We can however compute the gradient
+        with respect to the evolution time:
+
+        .. code-block:: python3
+
+            dev = qml.device("default.qubit", wires=2)
+
+            @qml.qnode(dev)
+            def my_circ(time):
+                # Prepare H:
+                H = qml.dot([0.2, -0.1], [qml.PauliY(0), qml.PauliZ(1)])
+
+                # Prepare some state
+                qml.Hadamard(0)
+
+                # Evolve according to H
+                qml.QDrift(H, time, n=10, seed=10)
+
+                # Measure some quantity
+                return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+
+        >>> args = qnp.array([1.23])
+        >>> print(qml.grad(my_circ)(*tuple(args)))
+        0.27980654844422853
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self, hamiltonian, time, n=1, seed=None, decomposition=None, id=None
```

Comment:
For some reason in `qml.RandomLayer` the random seed is set to a number (42). I'm not sure if this was done for any particular reason

Reply:
42 is just a common number in the community to set the seed. I think None is fine, we don't want to force a seed and have the user sample the same gates for every fresh instance of QDrift